vars/buildInfoYaml.groovy

```diff
@@ -0,0 +1,54 @@
+def call(Map args = [:]) {
+
+    def inputManifest = readYaml(file: args.inputManifest)
+    def outputFile = args.outputFile
+    def components = args.componentName
+    def componentsList = []
+    def status = args.status
+    echo("The status is $status")
+    echo("Components is $components")
+    if (!components.isEmpty()) {
+        echo ("Components parameter is not null")
+        for (component in components.split(" ")) {
+            componentsList.add(component.trim())
+        }
+    } else {
+        echo ("Components parameter is null")
+        inputManifest.components.each { component ->
+            componentsList.add(component.name)
+        }
+    }
+    echo (componentsList.toString())
+
+    if (args.stage == "START") {
```

Comment:
The `start` and `complete` should just be separate functions. Like `buildInfoCreate`, `buildInfoUpdate` and `buildInfoUpload`.

Reply:
After I discussed with @peterzhuamazon, we cut the the scope of this PR to exclude the status for individual stages. I think having separate groovy function can make it more complicate and hard to track.